lms/models/reporting/org_events.py

```diff
@@ -0,0 +1,115 @@
+import sqlalchemy as sa
+import logging
+from sqlalchemy import select, func, column
+from sqlalchemy.dialects.postgresql import insert
+from sqlalchemy import PrimaryKeyConstraint
+
+
+from lms.db import BASE
+from lms.models import Organization, Event, ApplicationInstance, EventUser, LTIRole
+
+LOG = logging.getLogger(__name__)
+
+
+class RollUpMixin:
+    rollup_index_elements = []
+    """Which columns are unique in the rollup table to find conflicts"""
+    rollup_query = None
+    """Query that produces the full row set of the rollup table"""
+    rollup_condition = None
+    """Condition to add to `rollup_query` when running this periodically"""
+
+    @classmethod
+    def rollup(cls, db_session, regenerate=False):
+        query = cls.rollup_query()
+        if not regenerate:
+            # Filter by the current day when updating periodically
+            query = query.where(cls.rollup_condition)
```

Comment:
The main assumption is that when using this, it much faster than a full materialized view refresh.